### PR TITLE
fix: Handle RFC names that are not all lowercase or have spaces

### LINKED ISSUE #4576: htmlizer entry point does wrong thing with cased inputs
From @agmalis

> I note that the URL
> 
> https://datatracker.ietf.org/doc/html/rfc7209
> 
> correctly displays RFC 7209.
> 
> However, the URL
> 
> https://datatracker.ietf.org/doc/html/RFC7209
> 
> actually displays draft-ietf-l2vpn-evpn-req-07.txt .
> 
> Cheers,
> Andy

### PR DESCRIPTION
**Pull request by larseggert**
Fixes #4576

### CODE REVIEW

File: ietf/doc/utils.py [hunk 1]
```diff
@@ -1293,6 +1293,7 @@ def fuzzy_find_documents(name, rev=None):
     document.
     """
     # Handle special case name formats
+    name = re.sub(r"\s+", "", name.lower())
```
Comment (rjsparks) [inline ietf/doc/utils.py:1296]: Can we restrict this to only happen if name.lower() startswith `rfc`?

File: ietf/doc/urls.py [hunk 2]
```diff
@@ -82,14 +82,14 @@
 
 # End of block that should be an idealized docs.ietf.org service instead
 
-    url(r'^html/(?P<name>[Rr][Ff][Cc] [0-9]+?)(\.txt|\.html)?/?$', views_doc.document_html),
+    url(r'^html/(?P<name>[Rr][Ff][Cc]\s*[0-9]+?)(\.txt|\.html)?/?$', views_doc.document_html),
```
Comment (rjsparks) [inline ietf/doc/urls.py:85]: What is the problem that led to wanting to do this? It changes the contract for what we'll provide, and we would be expected to keep this working in perpetuity. 
If it's a linkifier figuring out what to do, why not make it the linkifiers problem to normalize the produced URL instead?

**Comment by rjsparks** [comment]
> Fixes https://github.com/ietf-tools/datatracker/issues/4576
This is a surprising fix for that bug. I think it is probably just avoiding a bug we have elsewhere in the code that should be addressed. Did you determine _why_ this ended up on the last draft instead of the RFC? I would have expected there to be code that would be touching that decision directly.

**Comment by larseggert** [comment]
The issue is that the `if re.match("^[Rr][Ff][Cc] [0-9]+$",name):` case in the original code doesn't hit, because of the space in the pattern. And even if it did, it would rewrite `name` incorrectly, because the replacement assumes a space exists.

**Review by rjsparks — APPROVED** [review]

**Pull request merged**